### PR TITLE
Hide userAgentData model

### PR DESCRIPTION
**Pull request by pilgrim-brave**
Resolves https://github.com/brave/brave-browser/issues/20499

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: browser/farbling/brave_navigator_useragent_farbling_browsertest.cc [hunk 1]
```diff
@@ -231,3 +233,15 @@ IN_PROC_BROWSER_TEST_F(BraveNavigatorUserAgentFarblingBrowserTest,
   auto off_ua_b2 = EvalJs(contents(), kUserAgentScript);
   EXPECT_EQ(off_ua_b.ExtractString(), off_ua_b2);
 }
+
+// Tests results of farbling user agent metadata
+IN_PROC_BROWSER_TEST_F(BraveNavigatorUserAgentFarblingBrowserTest,
+                       FarbleNavigatorUserAgentModel) {
+  std::string domain_b = "b.com";
```
Comment (goodov) [inline browser/farbling/brave_navigator_useragent_farbling_browsertest.cc:240]: `domain_b` is not used anywhere, "b.com" should be passed directly to `GetURL`.

File: chromium_src/content/common/user_agent.cc [hunk 2]
```diff
@@ -8,6 +8,16 @@
 #define BRAVE_GET_ANDROID_OS_INFO \
   include_android_model = IncludeAndroidModel::Exclude;
 
+namespace content {
+std::string BuildModelInfo() {
+  std::string model;
+  return model;
```
Comment (goodov) [inline chromium_src/content/common/user_agent.cc:14]: `return std::string();`

File: chromium_src/content/common/user_agent.cc [hunk 3]
```diff
@@ -8,6 +8,16 @@
 #define BRAVE_GET_ANDROID_OS_INFO \
   include_android_model = IncludeAndroidModel::Exclude;
 
+namespace content {
```
Comment (goodov) [inline chromium_src/content/common/user_agent.cc:11]: can you put this block before or after all defines please? will be a little cleaner this way.

**Review by goodov — APPROVED** [review]
++ with few nits.

File: chromium_src/content/common/user_agent.cc (hunk 3 above)
Comment (pilgrim-brave) [reply chromium_src/content/common/user_agent.cc:8]: fixed in https://github.com/brave/brave-core/pull/11932/commits/2b48bd0b8e8e77ddb595fafb3f406b51e51d54a5

File: chromium_src/content/common/user_agent.cc (hunk 2 above)
Comment (pilgrim-brave) [reply chromium_src/content/common/user_agent.cc]: Fixed in https://github.com/brave/brave-core/pull/11932/commits/2b48bd0b8e8e77ddb595fafb3f406b51e51d54a5

File: browser/farbling/brave_navigator_useragent_farbling_browsertest.cc (hunk 1 above)
Comment (pilgrim-brave) [reply browser/farbling/brave_navigator_useragent_farbling_browsertest.cc]: Fixed in https://github.com/brave/brave-core/pull/11932/commits/2b48bd0b8e8e77ddb595fafb3f406b51e51d54a5

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 12` using `1.37.17 Chromium: 98.0.4758.87`

**`1.34.81 Chromium: 97.0.4692.99`** | **`1.37.17 Chromium: 98.0.4758.87`**
-----------------------------------------|--------------------------------------------
Screenshot_20220202-121837 | Screenshot_20220202-121857

**Pull request merged**